### PR TITLE
[SDKS-7493] Flagsets -  Add telemetry

### PR DESCRIPTION
**Pull request by emmaz90**
# Javascript commons library

## What did you accomplish?
 - Fix `validFilters` property in `splitFiltersValidation` & related tests
 - Change warning log for when sets and names filters are configured to log it as error
 - Add `originalFilter` property to `splitFiltersValidation` to keep raw original filters
 - Add `getTelemetryFlagsetsStats` function to get the amount of configured and ignored flagsets
 - Add fsT and fsI telemetry properties

## How do we test the changes introduced in this PR?
 - Unit tests included

## Extra Notes

### CODE REVIEW

File: src/sync/submitters/telemetrySubmitter.ts [hunk 1]
```diff
@@ -38,6 +39,25 @@ function getRedundantActiveFactories() {
   }, 0);
 }
 
+function getTelemetryFlagSetsStats(splitFiltersValidation: ISplitFiltersValidation) {
```
Comment (EmilianoSanchez) [inline src/sync/submitters/telemetrySubmitter.ts:42]: Having an internal `originalSets` array is unnecessary. We can reduce code by counting the number of total set filters in a reducer:

```
function getTelemetryFlagSetsStats(splitFiltersValidation: ISplitFiltersValidation) {
  const flagSetsTotal = splitFiltersValidation.validFilters.reduce((accum: number, filter: SplitIO.SplitFilter) => {
    return accum + (filter.type === 'bySet' ? filter.values.length : 0);
  }, 0);

  const flagSetsValid = splitFiltersValidation.groupedFilters.bySet.length;
  const flagSetsIgnored = flagSetsTotal - flagSetsValid;
  return { flagSetsTotal, flagSetsIgnored };
}
```

File: src/dtos/types.ts [hunk 2]
```diff
@@ -209,5 +209,6 @@ export interface IMetadata {
 export type ISplitFiltersValidation = {
   queryString: string | null,
   groupedFilters: Record<SplitIO.SplitFilterType, string[]>,
-  validFilters: SplitIO.SplitFilter[]
+  validFilters: SplitIO.SplitFilter[],
+  originalFilters: SplitIO.SplitFilter[]
```
Comment (EmilianoSanchez) [inline src/dtos/types.ts:213]: @emmaz90 , as we discussed earlier, the `originalFilters` object seems that is not necessary. We can use the `validFilters` array to count the total number of flag sets. 
Please, roll back the changes. Thank you :)

**Review by EmilianoSanchez — CHANGES_REQUESTED** [review]

**Review by EmilianoSanchez — APPROVED** [review]

File: src/sync/submitters/telemetrySubmitter.ts (hunk 1 above)
Comment (emmaz90) [reply src/sync/submitters/telemetrySubmitter.ts:42]: great catch doc! thank you!

File: src/dtos/types.ts (hunk 2 above)
Comment (emmaz90) [reply src/dtos/types.ts]: Rolled back and fixed validFilters beaviour 👏  thank you!

**Pull request merged**